### DISCUSSION TITLE
How to assert and click on secured iframe?

### DISCUSSION THREAD

**Discussion by SimonasB88** [cypress-io/cypress#23285] [Questions and Help]
Hey,
Ran into the issue, where the test code should click the button Process in the iframe. Used `npm i cypress-iframe` lib, but came up to nothing. Cypress could not find the button.

![image](https://user-images.githubusercontent.com/29095235/184158117-3657f153-ea50-467f-941d-234c084f6dd8.png)

`cy.iframe('[class="resp-iframe"]').find('resp-iframe[id="submit"]')`

also tried

```
cy.get('iframe[class="resp-iframe"]').then($element => {
             const $body = $element.contents().find('body')
             cy.wrap($body).find('resp-iframe[class="btn btn-block btn-primary"]').eq(0).click();
         })
```
also

```
cy.get('[class="resp-iframe"]').then($element => {
            const $body = $element.contents().find('body')
            let stripe = cy.wrap($body)
            stripe.find('[class="resp-iframe"]').click(150,150)
          })
```
Could someone check this out? Thanks.

**Comment by SimonasB88** [comment]
Maybe it is possible to click the button not with iframe?

**Reply by jjhelguero** [reply]
Does your get show the `cy.get('iframe[class="resp-iframe"]')` confirm you are getting the iframe you want?

**Reply by SimonasB88** [reply]
Yep, this iframe is asserted in UI:
![image](https://user-images.githubusercontent.com/29095235/185966153-19188a91-435c-4c4e-a297-33b462362c51.png)

Can not get into the right iframe level, to click on the Process:
![image](https://user-images.githubusercontent.com/29095235/185966449-5ce1f6d7-d781-4768-9bff-4b3bd2762465.png)

![image](https://user-images.githubusercontent.com/29095235/185967569-ea1259c3-f54d-47f1-aa43-a101c11f4d94.png)

**Comment by Priya-LaunchIT** [comment]
Can try this..

cy.frameLoaded('iframe')
cy.iframe().find('[id="submit"]').click()
